### PR TITLE
Smart Wallet Factory class for creating accounts and fetching data about them.

### PR DESCRIPTION
**Pull request by nkrishang**

### CODE REVIEW

File: packages/sdk/src/evm/core/classes/smart-wallet-factory.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+   * @twfeature SmartWalletFactory
+   */
+  public async getAllWallets(): Promise<AccountEvent[]> {
+    const filter = {
+      fromBlock: 0,
+      toBlock: "latest",
+    }
+
+    const events = await this.events.getEvents("AccountCreated", filter);
+    console.log(events);
```
Comment (jnsdls) [inline packages/sdk/src/evm/core/classes/smart-wallet-factory.ts:111]: can we remove the console log?

File: packages/sdk/src/evm/core/classes/smart-wallet-factory.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+   * 
+   * @returns all wallets created via the smart wallet factory.
+   * 
+   * @twfeature SmartWalletFactory
+   */
+  public async getAllWallets(): Promise<AccountEvent[]> {
+    const filter = {
+      fromBlock: 0,
+      toBlock: "latest",
+    }
```
Comment (jnsdls) [inline packages/sdk/src/evm/core/classes/smart-wallet-factory.ts:108]: this very well may fail in RPCs but we can try

Comment (nkrishang) [reply packages/sdk/src/evm/core/classes/smart-wallet-factory.ts:108]: I thought we could use the ethers [`deployTransaction.blockNumber`](https://docs.ethers.org/v5/api/contract/contract/#Contract--properties) api to set the `fromBlock` but that wouldn't work due to this requirement:

```
If the Contract object is the result of a ContractFactory deployment, this is the transaction which was used to deploy the contract.
```

Comment (joaquim-verges) [reply packages/sdk/src/evm/core/classes/smart-wallet-factory.ts:108]: @adam-maj how far are we from being able to plug specific events into our RPC / dedicated endpoint pulling data from chainsaw you think?

File: .changeset/poor-lions-retire.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@thirdweb-dev/contracts-js": minor
+"@thirdweb-dev/sdk": minor
```
Comment (joaquim-verges) [inline .changeset/poor-lions-retire.md:3]: this should be patch version change, not minor

File: packages/sdk/src/evm/contracts/smart-contract.ts [hunk 4]
```diff
@@ -292,6 +295,14 @@ export class SmartContract<
     return assertEnabled(this.detectOffers(), FEATURE_OFFERS);
   }
 
+  // TODO: documentation comments.
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/contracts/smart-contract.ts:298]: docs or remove?

**Pull request merged**